app/views/components/docs/email-signup-section.yml

```diff
@@ -0,0 +1,39 @@
+name: Email signup section
+description: Renders a box with a link to sign up for email notifications
+accessibility_criteria: |
+  - the component must use the correct heading level for the page
+  - text should have a text contrast ratio higher than 4.5:1 against the background colour to meet WCAG AA
+  - the icon must not be focusable or shown to screenreaders
+shared_accessibility_criteria:
+  - link
+examples:
+  default:
+    data:
+      intro: 'Sign up for email notifications'
+      link_text: 'Click right here to sign up!!'
+      link_href: '/this-signs-you-up'
+  with_background_and_border:
+    data:
+      intro: 'Sign up for email notifications'
+      link_text: 'Click right here to sign up!!'
+      link_href: '/this-signs-you-up'
+      background: true
+  link_only:
+    description: If no heading text is passed through, a more compact, link only version is rendered.
+    data:
+      link_text: 'Click right here to sign up!!'
+      link_href: '/this-signs-you-up'
+  with_custom_margin_bottom:
+    description: Override default `margin-bottom` of `0` by passing through a unit from the <a href="https://design-system.service.gov.uk/styles/spacing/">GOVUK responsive spacing scale</a>
```

Comment:
I've been trying to keep descriptions for this consistent (when components use the shared helper). Suggestion:

`The component accepts a number for margin bottom from 0 to 9 (0px to 60px) using the [GOV.UK Frontend spacing scale](https://design-system.service.gov.uk/styles/spacing/#the-responsive-spacing-scale). It defaults to having no margin bottom.`